src/test/groovy/graphql/normalized/ENOToAstCompilerDirectivesTest.groovy

```diff
@@ -0,0 +1,169 @@
+                [fooArgVar: "fooArgVar", barArgVar: "barArgVar", skipVar: false, optVar: ["optToY"]])
+
+        when:
+        def result = localCompileToDocument(schema, QUERY, "named",
+                eno.getTopLevelFields(), eno.getNormalizedFieldToQueryDirectives(),
+                allVariables)
+        def document = result.document
+        def vars = result.variables
+        def ast = printDoc(document)
+
+        then:
+        vars == [v0: "barArgVar", v1: ["optToX"], v2: ["optToY"], v3: "fooArgVar", v4: false]
+        //
+        // the below is what ir currently produces but its WRONG
+        // fix up when the other tests starts to work
+        //
+        ast == '''query named($v0: String, $v1: [String!]!, $v2: [String!]!, $v3: String, $v4: Boolean!) {
+  foo(fooArg: $v3) @skip(if: $v4) {
+    bar(barArg: $v0) @optIn(to: $v1) @optIn(to: $v2)
+  }
+}
+'''
+
+
+        when: "it has no variables"
+
+
+        result = localCompileToDocument(schema, QUERY, "named",
+                eno.getTopLevelFields(), eno.getNormalizedFieldToQueryDirectives(),
+                noVariables)
+        document = result.document
+        vars = result.variables
+        ast = printDoc(document)
+
+        then:
+        vars == [:]
+        ast == '''query named {
+  foo(fooArg: "fooArgVar") @skip(if: false) {
+    bar(barArg: "barArgVar") @optIn(to: ["optToX"]) @optIn(to: ["optToY"])
+  }
+}
+'''
+
+    }
+
+    def "can handle quite a pathological fragment query as expected"() {
+        def sdl = '''
+        directive @timeout(afterMillis : Int) on FIELD | FRAGMENT_DEFINITION | FRAGMENT_SPREAD | INLINE_FRAGMENT | QUERY
+        
+        directive @cached(forMillis : Int) on FIELD | FRAGMENT_DEFINITION | FRAGMENT_SPREAD | INLINE_FRAGMENT | QUERY
+        
+        directive @importance(place : String) on FIELD | FRAGMENT_DEFINITION | FRAGMENT_SPREAD | INLINE_FRAGMENT | QUERY
+ 
+        type Query {
+            books(searchString : String) : [Book]
+        }
+        
+        type Book {
+         id :  ID
+         title : String
+         review : String
+        }
+    '''
+
+        def pathologicalQuery = '''
+        fragment Details on Book @timeout(afterMillis: 25) @cached(forMillis : 25) @importance(place:"FragDef") {
+            title
+            review @timeout(afterMillis: 5) @cached(forMillis : 5)
+            ...InnerDetails @timeout(afterMillis: 26) 
+        }
+        
+        fragment InnerDetails on Book  @timeout(afterMillis: 27) {
+            review @timeout(afterMillis: 28)
+        }
+        
+        query Books @timeout(afterMillis: 30) @importance(place:"Operation") {
+            books(searchString: "monkey") {
+                id
+                 ...Details @timeout(afterMillis: 20)
+                 ...on Book @timeout(afterMillis: 15) {
+                    review @timeout(afterMillis: 10) @cached(forMillis : 10)
+                }
+            }
+        }
+    '''
```

Comment:
notice all the fragment directives

` fragment InnerDetails on Book  @timeout(afterMillis: 27)`

well they dont get picked up as "immediate" query directives.  This is NOT new behavior

Only immediate field directives are captured